nixos/modules/misc/ids.nix

```diff
@@ -515,7 +515,7 @@
       sddm = 175;
       tss = 176;
       #memcached = 177; # unused, removed 2018-01-03
-      #ntp = 179; # unused
+      ntp = 179;
```

Comment:
Is this necessary? Can't the module use DynamicUsers?

Reply:
I don't think we can use DynamicUsers here. This is not a new module, so it must maintain interoperability with configurations out there. While NTP might seem like it would not need to write anything to disk, it does. So changing the uid now would make NTP unable to read those files.

Also, this would be the first module in nixpkgs to use DynamicUsers.  I don't want to break the mold on my first MR.

To the question if this is necessary, as this is for the gid and not the uid anyway. I believe it to be good hygiene. NTP does write to disk, and under the old configuration those files would be owned by ntp:nogroup. This both does not look good, and grants other unprivileged "nogroup" services extra surface area on files ntp writes.

I have verified that changing the gid for NTP will not break anything as it also owns all the files it is using.

Reply:
It seems that a lot of modules are using `DynamicUser` (notice the missing s at the end) already:

```
$ git grep 'DynamicUser = true' | wc -l
45
```

Do those files that NTP writes need to persist across restarts or are they temporary files? Because if they are then it wouldn't hurt losing them.

Maybe you could also use `StateDirectory`, ` RuntimeDirectory`or `tmpfiles.d` or the likes for the temp files instead of `/var/lib/ntp`. See https://www.freedesktop.org/software/systemd/man/tmpfiles.d.html and https://www.freedesktop.org/software/systemd/man/systemd.exec.html#RuntimeDirectory=

Reply:
You are right about current usage... I am really getting burned by the letter `s` this MR. I did the grep for `DynamicUsers`. However my other points stand.

These files are not temporary, for ntpd this is namely the drift file is used.
>The drift file is used to store the frequency offset between the system clock running at its nominal frequency and the frequency required to remain in synchronization with UTC. If present, the value contained in the drift file is read at system start and used to correct the clock source. Use of the drift file reduces the time required to achieve a stable and accurate time. The value is calculated, and the drift file replaced, once per hour by ntpd. The drift file is replaced, rather than just updated, and for this reason the drift file must be in a directory for which ntpd has write permissions. 

For ntpsec there are can be several stats files that are meant to be read by other programs.

But beyond that, this is a large change to an existing core NTP module. If we want to go down the route to moving to DynamicUser then it should be in it's own CR, not complicate this one.

Reply:
I have removed the group bits here, which trims down the changes to the module. I am not opposed to switching to `DynamicUser` and making the service unit file much more explicit about what it can and cannot do. However the main goal of this MR is just adding NTPsec.

After it get's merged I'll create a MR for reworking this module a bit. But unlike this change of just adding a package, that change will require significantly more testing.